pkg/cri/server/container_create_linux.go

```diff
@@ -166,10 +166,13 @@ func (c *criService) containerSpec(
 
 	// Apply envs from image config first, so that envs from container config
 	// can override them.
-	env := append([]string{}, imageConfig.Env...)
+	env := []string{}
 	for _, e := range config.GetEnvs() {
 		env = append(env, e.GetKey()+"="+e.GetValue())
```

Comment:
Not new in this PR, but one thing I noticed here is that this code is _never_ able to _remove_ an env-var. `GetValue()` always returns a string, also if the value is `nil`. There's no distinction between `FOO=` (set `FOO` to an empty string) and `FOO` (`FOO` without value), so this code would never be hit; https://github.com/containerd/containerd/blob/23f66ece59654ea431700576b6020baffe1a4e49/oci/spec_opts.go#L197-L203

Reply:
The following loop will remove the empty-value env. If I understand correctly, the `FOO=` will be removed before return.
```
       // Now remove all entries that we want to "unset"
	for i := 0; i < len(results); i++ {
		if results[i] == "" {
			results = append(results[:i], results[i+1:]...)
			i--
		}
	}
```